### DISCUSSION TITLE
Feature: enforce document structure

### DISCUSSION THREAD

**Discussion by howtopick** [TypeCellOS/BlockNote#2426] [Ideas & Enhancements]
I try to fix `blockContainer+ ` to `heading blockContainer+` in BlockGroup.ts, but it don't work. someone know that?

**Comment by YousefED** [comment]
@howtopick this is not supported at the moment, we can look into it if more users require this!

**Comment by wenerme** [comment]
@YousefED Hi, I'd like to override the Doc schema like this https://tiptap.dev/docs/editor/examples/custom-document

I found this https://github.com/TypeCellOS/BlockNote/blob/main/packages/core/src/pm-nodes/Doc.ts

I hope I can change the schema to `heading blockGroup`

**Comment by sb8244** [comment]
I'll put my vote on this as well. I'd love to unify my title bar into the editor to be more like Notion

**Comment by deanthing** [comment]
Would love this feature.

We should be able to enforce document schema.
eg: All documents should start with an h1.

**Comment by woutervroege** [comment]
linking product goal: https://github.com/TypeCellOS/Product/issues/20
